nautobot/extras/utils.py

```diff
@@ -399,3 +399,49 @@ def refresh_job_model_from_job_class(job_model_class, job_source, job_class, *,
     )
 
     return (job_model, created)
+
+
+def migrate_role_data(
+    model,
+    role_model=None,
+    is_choice_field=False,
+    role_choiceset=None,
+    legacy_role="legacy_role",
+    new_role="new_role",
+):
+    """
+    Migrate legacy_role's `role_model` equivalent record into new_role.
+
+    Args:
+        model(Model): Model with role field to alter
+        role_model(Model): Role Model (optional)
+        is_choice_field(bool): True if Model's role field is a choice field not an FK field else False
+        role_choiceset(ChoiceSet): Role ChoiceSet (optional)
+        legacy_role(str): Models role field legacy name; This is the current role field name
+        new_role(str): Models role field new name; This is the name role field should be updated to
+    """
+
+    legacy_role_name = legacy_role if is_choice_field else legacy_role + "__name"
+    queryset = model.objects.filter(**{legacy_role + "__isnull": False}).values("pk", legacy_role_name)
```

Comment:
Is there actually a case with null `legacy_role` that we need to handle? Or is this just overly cautious?

Reply:
Yr apart from `Device` other model role fields allow null